### PR TITLE
document prebuilt workflow tasks

### PR DESCRIPTION
**Pull request by ydayagi**
flpath-402 https://issues.redhat.com/browse/FLPATH-402

### CODE REVIEW

File: documentation/latest/operations/prebuilt.md [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+- GitCommitTask
+- GitPushTask
+
+User can, optionally, provide Git credentials.
+
+## JDBC CRUD operation - JdbcWorkFlowTask
+
+The task uses JDBC for interacting with a database.
+The operations that the task executes are divided into three types.
+The type depends on the commands/statement that user provides.
```
Comment (gciavarrini) [inline documentation/latest/operations/prebuilt.md:42]: maybe `s/commands/command/` ?

File: documentation/latest/operations/prebuilt.md [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+
+- query. E.g. SQL select
+- update. E.g. SQL delete, insert, update
+- execute. E.g. SQL drop table, create table
+
+The user provides the JDBC connection URL. In the URL various
+parameters can be included.
+Username, password, TLS etc. Each JDBC driver has its own specific parameters.
```
Comment (gciavarrini) [inline documentation/latest/operations/prebuilt.md:50]: Should `Username, password, TLS etc.` be part of the previous sentence?

File: documentation/latest/operations/prebuilt.md [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+
+## Jira issue operation - JiraTask
+
+You can create, update or retrieve a Jira issue.
+Only basic fields are supported. E.g. description and comments.
```
Comment (gciavarrini) [inline documentation/latest/operations/prebuilt.md:55]: Can a list of all the supported fields or a link to the code be helpful to know exactly which fields the user can use?

File: documentation/latest/operations/prebuilt.md [hunk 4]
```diff
@@ -0,0 +1,234 @@
...
+In order to connect to the Jira server you will need to supply the
+URL and a bearer token.
+
+## Kubeapi CRUD operation - KubeapiWorkFlowTask
+
+User can create, delete, update and retrieve a Kubernetes resource.
+The user supplies a kubeconfig string for authentication and connection details.
+The resource is provided and retrieved in JSON format.
+You can find an example for Tekton resources at the end.
```
Comment (gciavarrini) [inline documentation/latest/operations/prebuilt.md:64]: I'd like to suggest using an anchor to the Tekton example to ease the doc browsing

File: documentation/latest/operations/prebuilt.md [hunk 5]
```diff
@@ -0,0 +1,234 @@
...
+## Notification service message - NotificationWorkFlowTask
+
+A task for sending a notification message to the notification service.
+
+## REST operations - RestWorkFlowTask
+
+A general purpose task for CRUD REST operations.
+User has control over the HTTP method and content of the request.
+For authentication, we support basic HTTP username and password authentication.
+In order to use TLS simply use the 'https://' scheme in the URL.
```
Comment (gciavarrini) [inline documentation/latest/operations/prebuilt.md:84]: nit:  use "`" (backticks) to properly format "http:/"

File: documentation/latest/operations/prebuilt.md [hunk 6]
```diff
@@ -0,0 +1,234 @@
...
+  },
+  "spec": {
+    "pipelineRef": {
+    "name": "pipeline-1"
+    }
+  }
+}
+```
+
+You will need a separate execution (request) for each resource.
```
Comment (gciavarrini) [inline documentation/latest/operations/prebuilt.md:175]: Do you think adding a ready-to-use request command would be helpful?

File: documentation/latest/operations/prebuilt.md (hunk 2 above)
Comment (ydayagi) [reply documentation/latest/operations/prebuilt.md:50]: there is a limit of 80 chars in line. so I have to split the lines. when displayed to user it looks as a single line.
image

File: documentation/latest/operations/prebuilt.md (hunk 3 above)
Comment (ydayagi) [reply documentation/latest/operations/prebuilt.md:55]: the user will have to go to the code anyhow. he will see it there.

File: documentation/latest/operations/prebuilt.md (hunk 5 above)
Comment (ydayagi) [reply documentation/latest/operations/prebuilt.md]: fixed

File: documentation/latest/operations/prebuilt.md (hunk 3 above)
Comment (gciavarrini) [reply documentation/latest/operations/prebuilt.md:55]: While it is true that users can refer to the code to see the supported fields, it is not always the most convenient or efficient method. Providing a list or a link can save users time and effort, especially for those who may not be familiar with the codebase or prefer a quick reference.

Additionally, including such information directly in the documentation can improve the overall user experience, making it easier for them to understand and utilize the functionality without having to navigate through the codebase.

However, I understand that everyone has their own perspective: feel free to proceed as you think best.

**Review by gciavarrini — APPROVED** [review]

**Pull request merged**